### PR TITLE
*: use FieldType.Decimal to control the decimal length of double values to be shown in client

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by zz-jason** [comment]
@XuHuaiyu @shenli PTAL

File: server/util_test.go [hunk 1]
```diff
@@ -52,37 +52,49 @@ func (s *testUtilSuite) TestDumpBinaryTime(c *C) {
...
+	bs, err := dumpTextValue(colInfo, types.NewIntDatum(10))
 	c.Assert(err, IsNil)
 	c.Assert(string(bs), Equals, "10")
 
-	bs, err = dumpTextValue(mysql.TypeLonglong, types.NewUintDatum(11))
+	bs, err = dumpTextValue(colInfo, types.NewUintDatum(11))
 	c.Assert(err, IsNil)
 	c.Assert(string(bs), Equals, "11")
 
+	colInfo.Type = mysql.TypeDouble
```
Comment (XuHuaiyu) [inline server/util_test.go]: TypeFloat ?

File: server/util_test.go [hunk 2]
```diff
@@ -52,37 +52,49 @@ func (s *testUtilSuite) TestDumpBinaryTime(c *C) {
...
 	c.Assert(string(bs), Equals, "1.2")
-	f32.SetFrac(2)
-	bs, err = dumpTextValue(mysql.TypeDouble, f32)
+
+	colInfo.Decimal = 2
+	bs, err = dumpTextValue(colInfo, f32)
 	c.Assert(err, IsNil)
 	c.Assert(string(bs), Equals, "1.20")
 
 	f64 := types.NewFloat64Datum(2.2)
```
Comment (XuHuaiyu) [inline server/util_test.go]: TypeDouble

**Comment by zz-jason** [comment]
updated, @XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
LGTM

File: server/util.go [hunk 3]
```diff
@@ -318,22 +318,22 @@ func dumpRowValuesBinary(alloc arena.Allocator, columns []*ColumnInfo, row []typ
 	return
 }
 
-func dumpTextValue(mysqlType uint8, value types.Datum) ([]byte, error) {
+func dumpTextValue(colInfo *ColumnInfo, value types.Datum) ([]byte, error) {
```
Comment (shenli) [inline server/util.go]: Should we also consider dumpRowValuesBinary()?

Comment (zz-jason) [inline server/util.go]: no need

**Comment by shenli** [comment]
LGTM

**Pull request merged**